### PR TITLE
Tb fix

### PR DESCRIPTION
**Pull request by cfmelend**
A quick fix to the tb TypeError issue seen [here](https://github.com/Noble-Lab/casanovo/issues/203). It seems like the `if not np.isnan(metric_value)` call in [this line](https://github.com/Noble-Lab/casanovo/blob/26b08cf4ba6f09bd64a2bbddb8c1975788cdbe4e/casanovo/denovo/model.py#L916C3-L916C3) is trying to convert some of the tensor values in metric_value (e.g. the train_loss) to numpy values. From what I can tell some of the values in metrics are floats and some are tensors so I added a short check to see if there are any tensors and if so make a copy of the variable, send to cpu, send to numpy (the last part might be wholly unnecessary).

### CODE REVIEW

File: casanovo/denovo/model.py [hunk 1]
```diff
@@ -913,10 +913,18 @@ def _log_history(self) -> None:
                     ("eval/val_aa_precision", "valid_aa_precision"),
                 ]:
                     metric_value = metrics.get(key, np.nan)
-                    if not np.isnan(metric_value):
-                        self.tb_summarywriter.add_scalar(
-                            descr, metric_value, metrics["step"]
-                        )
+                    if metric_value is not None:
```
Comment (bittremieux) [inline casanovo/denovo/model.py:916]: **suggestion:** This seems a bit overkill. Does it also work by doing `new_style=True` in [`set_scalar(...)`](https://pytorch.org/docs/stable/tensorboard.html#torch.utils.tensorboard.writer.SummaryWriter.add_scalar)?

Comment (cfmelend) [reply casanovo/denovo/model.py:916]: This works but it seemed to suppress logging output for the val/train loss. Tensorboard seems to work correctly on dev branch.
